strawberryfields/utils/post_processing.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module provides functions for post-processing samples.
+"""
+import numpy as np
+
+
+def _product_for_modes(samples, modes=None):
+    """Getting the product of samples across modes.
+Args:
+        samples (ndarray): the photon number samples with a shape
+            of (shots, modes)
+        modes (Sequence): a flat sequence containing indices of modes to get
+            the expectation value for
```

Comment:
This suggestion was made primarily to fix the indentation, but a couple of other comments:

* The shape should be given using code formatting
* I don't think it's necessary to say 'flat sequence'; this is a given when we say the type is
   ``Sequence``.

```suggestion
    """Getting the product of samples across modes.
        
    Args:
        samples (ndarray): the photon number samples with a shape of ``(shots, modes)``
        modes (Sequence): indices of modes to compute the expectation value for
```